src/main/java/com/team701/buddymatcher/controllers/users/UserController.java

```diff
@@ -290,4 +290,17 @@ public UserController(UserService userService, TimetableService timetableService
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
         }
     }
+
+    @Operation(summary = "Delete method for unblocking a user")
+    @DeleteMapping(path = "/unblock/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> unblockBuddy(@Parameter(hidden = true)
+                                             @SessionAttribute("UserId") Long userBlockerId,
+                                             @PathVariable("id") Long userBlockedId) {
+        try {
+            userService.unblockBuddy(userBlockerId, userBlockedId);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (NoSuchElementException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
```

Comment:
NoSuchElementException is not caught when the user is not found - is not in the database

Reply:
What do you mean by this? There is a catch statement there for catching the NoSuchElementException.

Reply:
It is not catching anything when a user is not found and 200 OK is being returned instead of 404 Not Found

Reply:
> What do you mean by this? There is a catch statement there for catching the NoSuchElementException.

It is not catching anything when a user is not found and 200 OK is being returned instead of 404 Not Found

Reply:
Are you sure about this? I can see that on line 302 the NoSuchElementException is getting caught and on line 303 ResponseStatusException is thrown with 404 NOT_FOUND status when NoSuchElementException is caught. Let me know if I'm reading this wrong.

Reply:
> Are you sure about this? I can see that on line 302 the NoSuchElementException is getting caught and on line 303 ResponseStatusException is thrown with 404 NOT_FOUND status when NoSuchElementException is caught. Let me know if I'm reading this wrong.

You could test it. Add a test case with a userId that doesn't exist in the database and the expectation that 404 Not Found is returned. 404 will not be returned.